### PR TITLE
Changed web default to no preference

### LINKED ISSUE #345: Change basic options default value from "exclude" to "no preference"
We should make it start with the one that's the least restrictive imo. It's also sort of weird to by-default exclude web-only classes, when that's like almost every class now

### PR DESCRIPTION
**Pull request by gannonprudhomme**
## Related tasks

Closes #345

### CODE REVIEW

**Comment by firejake308** [comment]
Can you also change the defaults in `autoscheduler/scheduler/utils.py`?

**Comment by gannonprudhomme** [comment]
Done!

**Comment by gannonprudhomme** [comment]
Ok it should be good to go now

File: autoscheduler/frontend/src/components/SchedulingPage/CourseSelectColumn/CourseSelectCard/ExpandedCourseCard/BasicSelect/BasicOptionRow.tsx [hunk 1]
```diff
@@ -16,7 +16,7 @@ interface BasicOptionRowProps {
  * option selected by this row, formatted as it is found in the Redux course cards
  */
 const BasicOptionRow: React.FC<BasicOptionRowProps> = ({ id, value }) => {
-  const option = useSelector<RootState, string>((state) => state.courseCards[id][value] || 'exclude');
+  const option = useSelector<RootState, string>((state) => state.courseCards[id][value] || 'no_preference');
```
Comment (ryan-conn) [inline autoscheduler/frontend/src/components/SchedulingPage/CourseSelectColumn/CourseSelectCard/ExpandedCourseCard/BasicSelect/BasicOptionRow.tsx:19]: This changes the default for honors as well, can you instead make it so the default is `no_preference` when a card is created with `addCourseCard`?

**Review by ryan-conn — CHANGES_REQUESTED** [review]

File: autoscheduler/frontend/src/components/SchedulingPage/CourseSelectColumn/CourseSelectCard/ExpandedCourseCard/BasicSelect/BasicOptionRow.tsx (hunk 1 above)
Comment (gannonprudhomme) [reply autoscheduler/frontend/src/components/SchedulingPage/CourseSelectColumn/CourseSelectCard/ExpandedCourseCard/BasicSelect/BasicOptionRow.tsx:19]: Oh yeah duh, this was the suggestion Adel gave. I was wondering why I hadn't done it previously when he suggested and this was the reason why.

**Review by ryan-conn — APPROVED** [review]
Third time's the charm lol, looks good

**Pull request merged**